### ISSUE TITLE
"Don't quote rhs of =~" should be a warning, not an error

### ISSUE THREAD

**Issue by pixarbuff** [koalaman/shellcheck#1985]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2076
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6
- [x ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
file=foo.csv
if [[ $file =~ ".csv" ]]; then echo yes; else echo no; fi

```

#### Here's what shellcheck currently says:
Line 3:
if [[ $file =~ ".csv" ]]; then echo yes; else echo no; fi
               ^-- SC2076: Don't quote right-hand side of =~, it'll match literally rather than as a regex.

#### Here's what I wanted or expected to see:
I agree that the WIKI already includes some advice related to using globbing rather than regex if matching literally, but if the syntax allows quoting to force literal matching, it should not come up as an error, but rather as a warning, if the quoted string contains regex metacharacters. 

The message should be a yellow warning rather than a red error, and should be reworded to match the other warnings, such as:
^-- SC2076: Remove quotes from right-hand side of =~ to match as a regex rather than literally.

**Comment by pixarbuff** [comment]
In Anlytics.hs please change
`err (getId t) 2076
                "Don't quote right-hand side of =~, it'll match literally rather than as a regex."`
to
`warn (getId t) 2076
                "Remove quotes from right-hand side of =~ to match as a regex rather than literally."`

**Comment by brother** [comment]
@pixarbuff I have no idea what I am doing really but it's more convenient to have it as a PR for the sake of discussion.

I put it in draft mode until more knowledge raises.

**Comment by pixarbuff** [comment]
Thanks, @brother

**Comment by brother** [comment]
Upgraded to a full PR to grab some attention.

**Comment by pixarbuff** [comment]
Thanks for your help! Could we do a similar thing with SC2068? It should be a warning, not an error: "Double quote array expansions to avoid re-splitting elements."
